### PR TITLE
HPCC-24728 Convert fileservices to use ecllib for windows installs

### PR DESCRIPTION
**Pull request by GordonSmith**
Removes the need to load the dll and choke on any further dependencies

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
There's a typo in the commit message

**Comment by GordonSmith** [comment]
@richardkchapman that was already fixed, but I updated the title here to reflect the change.

**Comment by richardkchapman** [comment]
@GordonSmith The smoketest failures need looking at

**Comment by GordonSmith** [comment]
Yes, I saw that.  May need to liaise with @ghalliday to work out why (and I would like to start building the client tools on my windows machine again).

File: plugins/proxies/lib_fileservices.ecllib [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+export FsLogicalSuperSubRecord := record string supername{maxlength(255)}; string subname{maxlength(255)}; end;
+export FsFileRelationshipRecord := record  string primaryfile   {maxlength(1023)}; string secondaryfile {maxlength(1023)}; string primaryflds   {maxlength(1023)}; string secondaryflds {maxlength(1023)}; string kind {maxlength(16)}; string cardinality   {maxlength(16)}; boolean payload; string description   {maxlength(1023)}; end;
+export integer4 RECFMV_RECSIZE := -2; // special value for SprayFixed record size
+export integer4 RECFMVB_RECSIZE := -1; // special value for SprayFixed record size
+export integer4 PREFIX_VARIABLE_RECSIZE := -3; // special value for SprayFixed record size
+export integer4 PREFIX_VARIABLE_BIGENDIAN_RECSIZE := -4; // special value for SprayFixed record size
+export FsDropZone := string;
+export FsDropZoneRecord := record FsDropZone dropzone; end;
+export FsLandingZoneRecord := record string name; string path; string hostname; end;
+export FileServices := SERVICE : plugin('fileserviceslib'), time
```
Comment (richardkchapman) [inline plugins/proxies/lib_fileservices.ecllib:33]: I think this should say "plugin('fileservices')"

**Review by richardkchapman — CHANGES_REQUESTED** [review]

**Comment by richardkchapman** [comment]
@GordonSmith I think you have the plugin name wrong in the ecllib file which is causing the setup errors

**Review by richardkchapman — APPROVED** [review]

**Comment by GordonSmith** [comment]
@ghalliday @richardkchapman - if the plugin loader only uses the ecllib file. "if" the dll does not include it, then this won't fix the underlying issue?

**Comment by ghalliday** [comment]
@GordonSmith this should fix the problem - because the system will not try to load the .dll file - so you will not hit the problem of the missing dependency.  You would still hit the problem if you built a local stand alone program and tried to run it - but that is fairly unusual.

**Review by ghalliday — APPROVED** [review]

**Pull request merged**